ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/validation/RequestValidator.java

```diff
@@ -105,6 +106,13 @@ public int validate(
     int numPersistentQueries = 0;
     for (ParsedStatement parsed : statements) {
       final PreparedStatement<?> prepared = ctx.prepare(parsed);
+
+      final Class<? extends Statement> statementClass = prepared.getStatement().getClass();
+      if (statementClass.equals(Query.class)
```

Comment:
I think a comment on why we're skip validation on pull queries would be helpful

Reply:
would `instance of` be a more robust check here? in case we decide to subclass Query down the line for e.g

Reply:
The `statementClass` is a subclass of `Statement`. `Query` is a subclass of `Statement` as well. There is no inheritance between them hence I cannot use `instance of`.

Reply:
`statementClass` is not a subclass of `Statement`, it is a `Class` instance. However, `prepared.getStatement()` returns a subtype of `Statement` and hence you can do:

```
if (prepared.getStatement() instance of Query.class) {
```
```